compiler/src/dmd/dtemplate.d

```diff
@@ -4129,6 +4128,37 @@ MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplateParameters* param
                                         goto L2;
                                 }
                             }
+                            if (TemplateDeclaration td = s.isTemplateDeclaration())
+                            {
```

Comment:
Can we merge the consecutive `if` statements since they have the same condition?
```suggestion

```

Reply:
4121 iters through the overloads of `td` and reassign `td` with them. Merging the two would require some modification between 4121 ~ 4130. I think we can revisit this later.

Reply:
You're right, not sure how I overlooked this.